tests/test_extractor.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+import json
+from pathlib import Path
+
+import pytest
+
+from qe_tools.extractors import extract
+
+
+@pytest.mark.parametrize('parser', ['pw', 'cp'])
+def test_input_dict_extraction(parser: str):
+    datadir = Path(__file__).resolve().parent / 'data'
+    filepath = datadir / f'{parser}.in'
+    refpath = datadir / 'extractor_ref' / f'{parser}.json'
+    with open(refpath, encoding='utf-8') as reference:
```

Comment:
Minor nitpick but since you use `pathlib.Path` you can just call `Path.open()`
```suggestion
    with refpath.open(encoding='utf-8') as reference:
```
Or better yet, if you are going to read it in memory simply do
```
assert extract(filepath.as_posix(), parser) == json.loads(refpath.read_text())
```

Reply:
Nice. Wasn't aware of this feature in `Path`. Thanks 🙏